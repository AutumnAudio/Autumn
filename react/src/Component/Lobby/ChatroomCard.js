import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper'
import { joinRoom } from '../AutumnApi/AutumnApi'
import {
    useHistory,
  } from "react-router-dom"
const styles = {
    paper: {
        height: 180,
        width: 180,
        background: 'pink',
    },
    chatroomName: {
        paddingTop: '65%',
        paddingRight: '10px',
        fontSize: '25px',
        color: 'white',
        textAlign: 'right',
    }
}
const joinChatroom = (genre, history) => {
    console.log(genre)
    joinRoom(genre).then((response) => {
        return response.text()
    }).then((text) => {
        console.log(text)
        history.push('/chatroom', {genre: genre})
    })
    
}
const getLengthOfObject = (object) => {
    let total = 0
    for (let i in object) {
        total++
    }
    return total
}
const ChatroomCard = (props) => {
    const { classes, chatroom } = props;
    const genre = chatroom.genre
    const history = useHistory()
    console.log(chatroom)
    const participantsNumber = getLengthOfObject(chatroom.participants)
    return (
        <Grid item>
            <Paper className={classes.paper} onClick={() => joinChatroom(genre, history)}>
                <div>Online: {participantsNumber}</div> 
                <div className={classes.chatroomName}>{chatroom.genre}</div>
            </Paper>
        </Grid>
    )
    
}
ChatroomCard.propTypes = {
    classes: PropTypes.object.isRequired,
    chatroom: PropTypes.object.isRequired,
};

export default withStyles(styles)(ChatroomCard)