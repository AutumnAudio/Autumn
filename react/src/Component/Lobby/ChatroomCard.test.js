import React from 'react'
import renderer from 'react-test-renderer'
import ChatroomCard from './ChatroomCard'

test('should be able to render chatroom card', () => {
    const chatroomMock = {chat: [],
                          genre: 'JAZZ',
                          link: '/joinroom/JAZZ',
                          participants: {'user1': {username: 'user1'}},
                          playlist: []}
    const component = renderer.create(
        <ChatroomCard chatroom={chatroomMock}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
