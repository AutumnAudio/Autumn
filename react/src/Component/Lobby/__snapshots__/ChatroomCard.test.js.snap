// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render chatroom card 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <div
    className="MuiPaper-root ChatroomCard-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
    onClick={[Function]}
  >
    <div>
      Online: 
      1
    </div>
    <div
      className="ChatroomCard-chatroomName-2"
    >
      JAZZ
    </div>
  </div>
</div>
`;
